### ISSUE TITLE
Make use of iOS 10 CallKit framework

### ISSUE THREAD

**Issue by jensrossbach** [wireapp/wire-ios#224]
Wire for iOS should make use of the new iOS 10 CallKit framework in order to provide a better telephony experience.

**Comment by mikeger** [comment]
Hey @jensrossbach ! Thanks for reporting the issue. We are looking into supporting the CallKit. It is actually already on our roadmap as one of the next things. However first we would have to port our sources to Swift 3.

**Comment by jenskasemets** [comment]
CallKit is now implemented in version 2.23, which is available in App Store

**Comment by jensrossbach** [comment]
@jenskasemets I tested the CallKit support with the release 2.23 but observed some problems:

1. Missed calls are not shown in lock screen and notification center.

2. When I start a video call, after accepting the call on iPhone, video is deactivated by default -> I have to press the video button, then unlock the phone, which redirects me automatically to the Wire app where I see the video picture of the caller (maybe this is by design / a limitation of iOS?).
Then there is a second problem: although video is indicated to be active (the video button is highlighted), it is actually still not (black screen with according notice on caller side). So I first have to deactivate video and activate it again in order to have it actually active.

3. I cannot make a call directly out of the iOS address book (via long press on the blue circular buttons on top of the contact screen); this is for instance offered for WhatsApp and Skype calls.

**Comment by jensrossbach** [comment]
@jenskasemets any comment on this? Should I create a new issue with these findings for tracking them?

Today I observed another issue. I called my iPhone from my iPad. Before accepting the call on the iPhone side, I hung up on iPad side. But the iPhone didn't stop ringing! After some time, I accepted this "incoming call". What happened was that a new call from iPhone to iPad was initiated. I stopped this on both sides. About 10 minutes later I received a notification on my iPhone about a missed call.
